lib/matplotlib/backend_tools.py

```diff
@@ -111,6 +117,23 @@ def destroy(self):
         """
         pass
 
+    def state_changed_event(self, sender, canvasevent, data):
```

Comment:
The name of the event implies an internal state, for non-toggled tools this is not True.